src/_pytest/deprecated.py

```diff
@@ -59,3 +61,10 @@
 STRICT_OPTION = PytestDeprecationWarning(
     "The --strict option is deprecated, use --strict-markers instead."
 )
+
+PRIVATE = PytestDeprecationWarning("A private pytest class or function was used.")
+
+
+def check_ispytest(ispytest: bool) -> None:
+    if not ispytest:
```

Comment:
is this cleaner with a sentinel object instead?

```
class IsPytest:
    pass

IS_PYTEST = IsPytest()

def check_ispytest(ispytest: IsPytest) -> None:
    if ispytest is not IS_PYTEST:
        warn(PRIVATE, stacklevel)
```

I don't see any `*, _ispytest: bool = True` default set, so I can't see how the PytestDeprecationWarning would be triggered

Reply:
Is the reason for the sentinel to make it even harder to bypass? TBH my intention is not to prevent anyone from passing `_ispytest=True` in their own code if they want to bypass the privacy check, this is pytest/python after all :) My intention is just to add a simple hurdle that says "warranty is off, this may break without notice".

Reply:
(Marking as resolved for now based on my reply)